arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/call/CallSection.java

```diff
@@ -361,8 +354,17 @@ private void smcProcessing(Hub hub, MessageFrame frame) {
   }
 
   private void prcProcessing(Hub hub) {
+    precompileSubsection =
+        ADDRESS_TO_PRECOMPILE.get(preOpcodeCalleeSnapshot.address()).apply(hub, this);
     hub.defers().scheduleForContextEntry(this);
     hub.defers().scheduleForContextReEntry(this, hub.currentFrame());
+    // In case of arguments too large for MODEXP, transaction will be popped anyway, and resolving
+    // some defers will create NPE
+    if (precompileSubsection instanceof ModexpSubsection
+        && ((ModexpSubsection) precompileSubsection).transactionWillBePopped) {
+      hub.defers().unscheduleForContextReEntry(this, hub.currentFrame());
+      hub.defers().unscheduleForPostTransaction(this);
+    }
```

Comment:
Wouldn't it make more sense to
- schedule all the `PrecompileSubsection`'s that aren't `instanceof ModexpSubsection`
- only schedule a `ModexpSubsection` if `!transactionWillBePopped` ?

You are scheduling and unscheduling but you already have the information as to whether you will need to `unschedule`.

Reply:
I do agree, but I find it way more readable to do it in any case, and undo it in some (very rare) case.

Reply:
plus when we schedule for postTransaction, we still don't have the information as the precompile subsection is created after. So in any case we'll have to unschedule it.